src/Core/src/Response.php

```diff
@@ -85,24 +86,74 @@ public function resolve(?float $timeout = null): bool
                 }
             }
 
-            $statusCode = $this->httpResponse->getStatusCode();
+            return $this->handleStatus();
         } catch (TransportExceptionInterface $e) {
             throw $this->resolveResult = new NetworkException('Could not contact remote server.', 0, $e);
         }
+    }
 
-        if (500 <= $statusCode) {
-            throw $this->resolveResult = new ServerException($this->httpResponse);
-        }
+    /**
+     * Make sure all provided requests are executed.
+     *
+     * @param self[]     $responses
+     * @param float|null $timeout   Duration in seconds before aborting. When null wait
+     *                              until the end of execution. Using 0 means non-blocking
+     *
+     * @return iterable<self>
+     *
+     * @throws NetworkException
+     * @throws HttpException
+     */
+    final public static function multiplex(iterable $responses, float $timeout = null): iterable
```

Comment:
I think it makes sense to add `$fullResponse` or similar to this method. But not the "resolve" function. Hm.. or am I wrong?

Reply:
good idea! I like it!
it avoid adding another parameter to `resolve` and focus multiplexing on that method only.